src/libraries/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -200,19 +200,33 @@ internal static TextReader GetOrCreateReader()
         // where the assumption of KeyDown-KeyUp pairing for a given key press
         // is invalid. For example in IME Unicode keyboard input, we often see
         // only KeyUp until the key is released.
-        private static bool IsKeyDownEvent(Interop.INPUT_RECORD ir)
+        private static bool IsReadKeyEvent(ref Interop.INPUT_RECORD ir)
         {
-            return (ir.EventType == Interop.KEY_EVENT && ir.keyEvent.bKeyDown != Interop.BOOL.FALSE);
-        }
-
-        private static bool IsModKey(Interop.INPUT_RECORD ir)
-        {
-            // We should also skip over Shift, Control, and Alt, as well as caps lock.
-            // Apparently we don't need to check for 0xA0 through 0xA5, which are keys like
-            // Left Control & Right Control. See the ConsoleKey enum for these values.
-            ushort keyCode = ir.keyEvent.wVirtualKeyCode;
-            return ((keyCode >= 0x10 && keyCode <= 0x12)
-                    || keyCode == 0x14 || keyCode == 0x90 || keyCode == 0x91);
+            if (ir.EventType != Interop.KEY_EVENT) return false;
+            if (ir.keyEvent.bKeyDown == Interop.BOOL.FALSE)
```

Comment:
From https://github.com/dotnet/runtime/blob/main/docs/coding-guidelines/coding-style.md: *Braces may be omitted only if the body of every block associated with an if/else if/.../else compound statement is placed on a single line.*

Please add braces to every if/else block in this method.

```suggestion
            if (ir.EventType != Interop.KEY_EVENT)
            {
                return false;
            }
            else if (ir.keyEvent.bKeyDown == Interop.BOOL.FALSE)
```